apps/ledger-live-mobile/src/newArch/features/WalletSync/components/Synchronize/ScanQrCode.tsx

```diff
@@ -72,17 +73,22 @@ const ScanQrCode = ({ onQrCodeScanned }: Props) => {
           justifyContent={"center"}
           alignItems={"center"}
         >
-          <CameraView
-            style={{
-              backgroundColor: colors.neutral.c50,
-              width: 280,
-              height: 280,
-            }}
-            barcodeScannerSettings={{
-              barcodeTypes: ["qr"],
-            }}
-            onBarcodeScanned={onBarCodeScanned}
-          />
+          <CameraPermissionContext.Consumer>
```

Comment:
I tried to define a component to avoid using `CameraPermissionContext.Consumer`:
```tsx
function WaitForPermissionCameraView(props: CameraProps) {
  const { permissionGranted } = useContext(CameraPermissionContext);
  return <CameraView active={permissionGranted ?? false} {...props} />;
}
```
But this implementation didn't solves the problem on iOS (even when adding a key changing based on the permission).